tests/integration/security/filebased_tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,527 @@
+        - gateways:
+            - istio-egressgateway
+          port: 80
+      route:
+        - destination:
+            host: server.{{.AppNamespace}}.svc.cluster.local
+            port:
+              number: 443
+          weight: 100
+      headers:
+        request:
+          add:
+            handled-by-egress-gateway: "true"
+`
+)
+
+// We want to test "external" traffic. To do this without actually hitting an external endpoint,
+// we can import only the service namespace, so the apps are not known.
+// If some service(client) in appNamespace wants to call another service in appNamespace(server) it cannot
+// directly call it as sidecarscope only allows traffic to serviceNamespace, Gateway in serviceNamespace
+// then comes into action and receives this "outgoing" traffic to only route it back into appNamespace(server)
+func createGateway(t *testing.T, ctx resource.Context, appsNamespace namespace.Instance, serviceNamespace namespace.Instance) {
+	tmpl, err := template.New("Gateway").Parse(Gateway)
+	if err != nil {
+		t.Fatalf("failed to create template: %v", err)
+	}
+
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, map[string]string{"AppNamespace": appsNamespace.Name()}); err != nil {
+		t.Fatalf("failed to create template: %v", err)
+	}
+	if err := ctx.ApplyConfig(serviceNamespace.Name(), buf.String()); err != nil {
+		t.Fatalf("failed to apply gateway: %v. template: %v", err, buf.String())
+	}
+}
+
+func clusterName(target echo.Instance, port echo.Port) string {
+	cfg := target.Config()
+	return fmt.Sprintf("outbound|%d||%s.%s.svc.%s", port.ServicePort, cfg.Service, cfg.Namespace.Name(), cfg.Domain)
+}
+
+// Wait for the server to NOT be callable by the client. This allows us to simulate external traffic.
+// This essentially just waits for the Sidecar to be applied, without sleeping.
+func WaitUntilNotCallable(c echo.Instance, dest echo.Instance) error {
+	accept := func(cfg *envoyAdmin.ConfigDump) (bool, error) {
+		validator := structpath.ForProto(cfg)
+		for _, port := range dest.Config().Ports {
+			clusterName := clusterName(dest, port)
+			// Ensure that we have an outbound configuration for the target port.
+			err := validator.NotExists("{.configs[*].dynamicActiveClusters[?(@.cluster.Name == '%s')]}", clusterName).Check()
+			if err != nil {
+				return false, err
+			}
+		}
+
+		return true, nil
+	}
+
+	workloads, _ := c.Workloads()
+	// Wait for the outbound config to be received by each workload from Pilot.
+	for _, w := range workloads {
+		if w.Sidecar() != nil {
+			if err := w.Sidecar().WaitForConfig(accept, retry.Timeout(time.Second*10)); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func mountCACerts(t *testing.T, ctx resource.Context) error {
+	systemNs, err := namespace.ClaimSystemNamespace(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeAccessor := ctx.Environment().(*kube.Environment).KubeClusters[0]
+	err = kubeAccessor.PatchDeployment(systemNs.Name(), "istio-egressgateway", `
```

Comment:
Any doc that has `kubectl patch` should be deleted, its horrible practice. So I think its good the test follows the doc, but the doc absolutely should not use kubectl patch. Instead we should do it as part of the proper install. It is configurable, see https://github.com/istio/istio/blob/c05e57fe55a3c801a21574c2f1f3b6c451246606/manifests/charts/gateways/istio-egress/templates/deployment.yaml#L241. So I would highly recommend updating the doc, and the test, to use that instead of patch.

Could be after this merges

Reply:
I might have to transition to that approach as it seems like either patch is not applying correctly during CI run or patch is applying but egress gw pod is reporting "ready" state right away which isn't giving time for certs to kick in

Reply:
there's already a PR:https://github.com/istio/istio.io/pull/6194 to change up the doc which is blocked by the ISTIO_MUTUAL issue. That aside I changed the mounting approach in this PR